### PR TITLE
Update CODEOWNERS to use OpenTelemetry team

### PR DESCRIPTION
**Pull request by MikeGoldsmith**
Updates the codeowners file to use the Github team @open-telemetry/ruby-approvers instead of discrete list of users. This will make managing the group easier as it doesn't require code changes when the approvers list is updated.

### CODE REVIEW

**Review by ahayworth — DISMISSED** [review]

**Comment by fbogsany** [comment]
While I appreciate the convenience of this from a maintenance perspective, is there a way to be notified of changes to the membership of @open-telemetry/ruby-approvers? Updating this file manually currently notifies anyone watching this repo.

Side note: your PR description refers to rust-approvers rather than ruby-approvers.

**Comment by MikeGoldsmith** [comment]
I'm not certain if Github teams have a notification feature on edit. Is it important to update all repository watchers of approver changes?

(I've updated the PR description to reference the correct type - must have been auto-complete error).

**Comment by fbogsany** [comment]
> Is it important to update all repository watchers of approver changes?

It's a nice side effect. Most discussion of approver changes happen in SIG meetings, which likely have a more limited audience than PRs to this repo. I know I would like to know if the approvers list was changed 😄 - this PR made me wonder how I would be informed of that.

**Comment by arielvalentin** [comment]
> It's a nice side effect. Most discussion of approver changes happen in SIG meetings, which likely have a more limited audience than PRs to this repo. I know I would like to know if the approvers list was changed 😄 - this PR made me wonder how I would be informed of that.

@fbogsany if I understand your need here you want to subscribe to team changed events? That way you could be notified is someone is added or removed from a team. Is that correct?

**Comment by robertlaurin** [comment]
This may not be the best justification for keeping it the way it is, but having an explicit code owners list creates a fun little ceremony when adding members to the group.  I see it as a positive thing, and given the final say I would keep it as is because it promotes the type of community I seek to be part of.

**Comment by robertlaurin** [comment]
Going to close this as I don't see us making this change in the near future.

**Comment by ericmustin** [comment]
long live _the ceremony_

**Pull request closed without merging**
